arkouda/groupbyclass.py

```diff
@@ -263,10 +263,41 @@ def __init__(
                 self.keys, return_groups=True, assume_sorted=self.assume_sorted
             )
 
-        self.size = self.permutation.size
+        self.length = self.permutation.size
         self.ngroups = self.segments.size
 
-    def count(self) -> Tuple[groupable, pdarray]:
+    @typechecked
+    def size(self) -> Tuple[Union[Tuple, pdarray, Strings], pdarray]:
```

Comment:
I think the typechecking for `size` and `count` should be the same as `count` was before i.e.
```python
def count(self) -> Tuple[groupable, pdarray]:
```

This will include the elements you have here `pdarray` and `Strings`. As well as `Categorical` and or other sequence of these (`Tuple`, `List`, ...)
https://github.com/Bears-R-Us/arkouda/blob/b6122cba6b267c2c12ce6c7d850b47c37e84391f/arkouda/groupbyclass.py#L28-L29

Reply:
When I tested it using the original type hints, I got an error that said Categorical was undefined.

Reply:
I just pulled down the PR and i see what you mean, we should still support `Categorical`. So let's restore the original type hints and remove `@typechecked`

I also think we should add a comment near the top (perhaps right under the definition of `groupable`) that says 
```
Note: we won't be typechecking GroupBy until we can figure out a way to handle the circular import with Categorical
```

Reply:
Done